src/apps/reservation-list/modal/reservation-details/reservation-details.tsx

```diff
@@ -6,27 +6,44 @@ import fetchMaterial, {
 import ModalDetailsHeader from "../../../../components/modal-details-header/modal-details-header";
 import DigitalListDetails from "./digital-list-details";
 import ReservationDetailsButton from "./reservation-details-buttons";
-import { AgencyBranch } from "../../../../core/fbs/model";
 import ReservationDetailsRedirect from "./reservation-details-redirect";
 import { useText } from "../../../../core/utils/text";
 import fetchDigitalMaterial from "../../../loan-list/materials/utils/digital-material-fetch-hoc";
 import PhysicalListDetails from "./physical-list-details";
+import { useConfig } from "../../../../core/utils/config";
+import { AgencyBranch } from "../../../../core/fbs/model";
+import { excludeBlacklistedBranches } from "../../../../core/utils/branches";
 
 export interface ReservationDetailsProps {
   reservation: ReservationType;
-  branches: AgencyBranch[];
+  openReservationDeleteModal: (deleteId: number | string | null) => void;
 }
 
 const ReservationDetails: FC<ReservationDetailsProps & MaterialProps> = ({
   reservation,
   material,
-  branches
+  openReservationDeleteModal
 }) => {
   const t = useText();
   const { state, identifier, numberInQueue } = reservation;
   const { authors, pid, year, title, description, materialType } =
     material || {};
+  const config = useConfig();
+  // Get library branches from config
+  const inputBranches = config<AgencyBranch[]>("branchesConfig", {
+    transformer: "jsonParse"
+  });
+
+  // Get the library branches where the user cannot pick up books at
+  const blacklistBranches = config("blacklistedPickupBranchesConfig", {
+    transformer: "stringToArray"
+  });
 
+  // Remove the branches where the user cannot pick up books from the library branches
+  let branches = inputBranches;
+  if (Array.isArray(blacklistBranches)) {
+    branches = excludeBlacklistedBranches(inputBranches, blacklistBranches);
+  }
```

Comment:
Kasper B. has made a hook that does something similar, so if you can use it I think we should not repeat the functionality. Please take a look at it:
https://github.com/reload/dpl-react/blob/cf64e43d25211f3a141e6ea195f031dc1c88d654/src/core/utils/branches.tsx#L26

Reply:
Jep ill change

Reply:
This doesn't work with a function I use later, https://github.com/danskernesdigitalebibliotek/dpl-react/blob/c61d078d0a51f5bc78991b929c28fdd4ef6869b5/src/components/reservation/helper.ts#L23 i'll add a todo for now